lib/tz_world.ex

```diff
@@ -1,89 +1,139 @@
+      {:ok, "Europe/Paris"}
+
+      iex> TzWorld.timezone_at(3.2, 45.32)
+      {:ok, "Europe/Paris"}
+
+
+  The algorithm starts by filtering out timezones whose bounding
+  box does not contain the given point.
+
+  Once filtered, the first timezone which contains the given
+  point is returned, or `nil` if none of the timzones match.
 
-  The algorithm starts by filtering out timezones whose bounding box does not contain
-  the given point.
-  Once filtered, we return the first timezone which contains the given point, or `nil` if
-  none of the timzones match.
   """
-  @spec timezone_at(Geo.Point.t) :: String.t | nil
+  @spec timezone_at(Geo.Point.t()) :: String.t() | nil
   def timezone_at(%Point{} = point) do
-    GenServer.call(__MODULE__, {:timezone_at, point})
+    GenServer.call(__MODULE__, {:timezone_at, point}, @timeout)
   end
 
-  def handle_info(:init_state, _state) do
-    {:noreply, read_data()}
+  @spec timezone_at(lng :: number, lat :: number) :: String.t() | nil
+  def timezone_at(lng, lat) when is_lng(lng) and is_lat(lat) do
+    point = %Geo.Point{coordinates: {lng, lat}}
+    GenServer.call(__MODULE__, {:timezone_at, point}, @timeout)
   end
 
-  defp read_data do
-    {:ok, [data_file]} = :zip.unzip(to_charlist(@data_archive))
-    data_file
-    |> File.read!()
-    |> :erlang.binary_to_term()
+  @doc """
+  Reload the timezone geo JSON data.
+
+  This allows for th data to be reloaded,
+  typically with a new release, without
+  restarting the application.
+
+  """
+  @spec reload_timezone_data :: :ok
+  def reload_timezone_data do
+    GenServer.call(__MODULE__, :reload_data, @timeout)
+  end
+
+  # --- Server callback implementation
+
+  def handle_continue(:load_data, _state) do
+    {:noreply, GeoData.load_compressed_data()}
+  end
+
+  def handle_call(:reload_data, _from, _state) do
+    case GeoData.load_compressed_data() do
+      {:ok, _data} = return -> {:reply, :ok, return}
+      other -> {:reply, other, other}
+    end
+  end
+
+  def handle_call(:version, _from, state) do
+    case state do
+      {:ok, [version | _tz_data]} -> {:reply, {:ok, version}, state}
+      other -> {:reply, other, state}
+    end
   end
 
   def handle_call({:timezone_at, %Geo.Point{} = point}, _from, state) do
     timezone =
-      state
-      |> Enum.filter(fn({_, _, bounding_box}) -> contains?(bounding_box, point) end)
-      |> Enum.find(fn({_, geometry, _}) -> contains?(geometry, point) end)
-      |> case do
-        {timezone, _, _} ->
-          timezone
-        nil ->
-          nil
+      with {:ok, [_version | tz_data]} <- state do
+        tz_data
+        |> Enum.filter(fn geometry -> contains?(geometry.properties.bounding_box, point) end)
+        |> Enum.find(&contains?(&1, point))
+        |> case do
+          %Geo.MultiPolygon{properties: %{tzid: tzid}} -> {:ok, tzid}
+          %Geo.Polygon{properties: %{tzid: tzid}} -> {:ok, tzid}
+          nil -> {:error, :notfound}
```

Comment:
I think that a more explicit `:timezone_not_found` error would be better. When I first saw it I was not really sure about what was not found (the timezone or the data file).